packages/common/core-utils/src/assert.ts

```diff
@@ -71,8 +71,29 @@ const firstChanceAssertionHandler = new Set<(error: Error) => void>();
  * @param handler - Called when an assertion occurs before the exception is thrown.
  * @returns a function to remove the handler.
  * @remarks
- * This is mainly useful for triggering a `debugger;` statement and/or reporting telemetry to aid Fluid with diagnosing and fixing the bugs which cause asserts.
- * It can also be used to record that an fatal error has occurred to help avoid complicating the diagnosis by doing more work after the error which might error again in a different way.
+ * The callback runs just before the exception is thrown which makes it a better place to report telemetry for Fluid Framework bugs than a catch block or an event like `window.onerror`.
+ * Using this API to report telemetry is preferred over over those approaches since it eliminates the risk of the exception being swallowed or obfuscated by an intermediate stack frame's catch block
```

Comment:
```suggestion
 * Using this API to report telemetry is preferred over those approaches since it eliminates the risk of the exception being swallowed or obfuscated by an intermediate stack frame's catch block
```